internals/metrics/metrics.go

```diff
@@ -63,28 +73,29 @@ func NewOpenTelemetryWriter(w io.Writer) *OpenTelemetryWriter {
 }
 
 func (otw *OpenTelemetryWriter) Write(m Metric) error {
-	var metricType string
-	switch m.Type {
-	case TypeCounterInt:
-		metricType = "counter"
-	case TypeGaugeInt:
-		metricType = "gauge"
+	if m.Comment != "" {
+		_, err := fmt.Fprintf(otw.w, "# HELP %s %s\n", m.Name, m.Comment)
+		if err != nil {
+			return err
+		}
 	}
 
-	_, err := fmt.Fprintf(otw.w, "# HELP %s %s\n", m.Name, m.Comment)
-	if err != nil {
-		return err
-	}
-	_, err = fmt.Fprintf(otw.w, "# TYPE %s %s\n", m.Name, metricType)
+	_, err := fmt.Fprintf(otw.w, "# TYPE %s %s\n", m.Name, m.Type.String())
 	if err != nil {
 		return err
 	}
 
-	labels := make([]string, len(m.Labels))
-	for i, label := range m.Labels {
-		labels[i] = fmt.Sprintf("%s=%s", label.key, label.value)
+	io.WriteString(otw.w, m.Name)
```

Comment:
Note that we're not checking errors here (but we are above). In this case that's actually okay given that we know the writer is a `bytes.Buffer` (which never fails), but for future-proofing and reliability, let's check each error here.

Reply:
Got it, I will check all errors for all `io.WriteString` and `fmt.Fprintf` to the writer.